docs/reference/method/MongoDBDatabase-renameCollection.txt

```diff
@@ -0,0 +1,76 @@
+===================================
+MongoDB\\Database::renameCollection()
+===================================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+Definition
+----------
+
+.. phpmethod:: MongoDB\\Database::renameCollection()
+
+   Changes the name of an existing collection.
+
+   .. code-block:: php
+
+      function renameCollection($fromNamespace, $toNamespace, array $options = []): array|object
```

Comment:
I don't think it works to determine whether a collection name or namespace is being provided by checking whether the argument contains a dot, since the collection names used in the tests contain a dot as well. So I'm not sure how it would be possible to do this actually - any ideas?

Reply:
Very good point. This would work the other way around (e.g. `$databaseNameOrNamespace`) because database names cannot contain dots; however, it's ambiguous if we're trying to differentiate a namespace from a collection name.

In that case, I think it makes sense to leave `$toNamespace` as-is.

`$fromNamespace` can still be changed to `$collectionName` for consistency with `Database::dropCollection()`, and the helper method can build the namespace to send to the Operation internally.

Reply:
To add on to this, let's change the API of both the helpers _and_ Operation class to no longer take namespace strings as paramaters:

```
$database->renameCollection($collectionName, $toDatabaseName, $toCollectionName);
$collection->rename($toDatabaseName, $toCollectionName);
```

The Operation class will then have four required parameters:

 * fromDatabaseName
 * fromCollectionName
 * toDatabaseName
 * toCollectionName

It can then build the namespace internally with string concatenation in the constructor (allowing you to keep the internal private properties that store namespaces).

For the Database helper, fromDatabaseName would be pulled from the database instance's own name. And for the Collection helper, both "from" parameters would be pulled from the collection instance.